SharedTests/DeferredTests.swift

```diff
@@ -68,4 +68,29 @@ class DeferredTests: XCTestCase {
 
         waitForExpectations(timeout: 10, handler: nil)
     }
+
+    func testAccumulate() {
+        let leak = self.expectation(description: "deinit")
+
+        class TestClass {
+            let end: XCTestExpectation
+            init(e: XCTestExpectation) {
+                end = e
+                accumulate([self.aSimpleFunction]).upon { _ in
+
+                }
+            }
+
+            func aSimpleFunction() -> Success {
+                return succeed()
+            }
+            deinit {
```

Comment:
> Does this exercise the bug because TestClass won't be deinited if one of its methods is alive as a thunk?

Yea the expectation would not be filled before this fix.

>If so, weak self in aSimpleFunction would make this test pass if the bug were still present, right?  

You cant make `aSimpleFunction` use a weak self. We could wrap the function in a closure that uses a weak self. But I tried that, Doesnt work 😢 

> Is there a way to write a test that really exercises the leak? 

The test should exercise the leak. When you `nil` the class you wait for the deinit call to happen in the test case. If you mean is there a way to make the leak happen with the fix applied. I dont know how I would do that. 

> you also need a little test for what happens if one of the thunks returns failure.
 
I'll add that 😄 

Thanks for taking time out of your holiday. Really appreciate it!